### PR TITLE
add readme

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

A rough readme to get something up and running. There are still things that are unclear but it'd give people an idea about what it does and how they can get started if they feel adventurous.

We'll need to do another pass in the coming weeks.

Fixes https://github.com/crossplane-contrib/terrajet/issues/32

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N/A

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+In Terrajet, we have code generators for generating the Go types of the resource
+from imported Terraform provider. Additionally, we have a common controller that
+implements `managed.ExternalClient`. Using `pipeline` package, you can generate
+the API types as well as the code that stitches it together with the implementation
+of `managed.ExternalClient`.
+
+The structure of the generator is just like any other Go program; we'll create
+a `cmd/generator/main.go` file and call Terrajet tools from there.
+
+The first step would be to use [`provider-template`][provider-template] to
```
Comment (luebken) [inline README.md:30]: From my understanding, there are generally 3 components here:

1) A terraform provider e.g. https://github.com/rancher/terraform-provider-rancher2/ 
2) The Provider template https://github.com/crossplane/provider-template
and
3) Terrajet https://github.com/crossplane-contrib/terrajet

Possibly obvious to others but could you explain how these three fit together and where I would add the code generator?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+This code excerpt is the main piece that you need to implement to get the code
+generators working. After the first run, you'll see many CRDs generated. In order
+to run other generic code generators, run the following command:
+```
+make generate
+```
+
+### Provider Config Builder
+
+<to be filled>
```
Comment (luebken) [inline README.md:164]: Could you describe / reference what Provider Config Builder is about?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+### Local Testing
+
+Once you complete the first section, you'll have many CRDs in `package/crds`. You
```
Comment (luebken) [inline README.md:168]: I love steps. Maybe we can layout the different sections above?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+At this point, the managed resources you create will be reconciled.
+
+### Packaging
+
+> This section might go away once we have a provider-tf-template or a branch
+> in provider-template that you can use as template for your Terraform-based
+> provider.
+
+Since the common controller shells out to Terraform CLI, which in turn talks to
+its provider plugin, we'd like to make those two available in the final controller
```
Comment (luebken) [inline README.md:181]: Non-blocking todo for later: We should have some kind of architectural overview on how these components work together. Especially the new components that come with Terraform.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+* Slack: Join our [slack channel](https://slack.crossplane.io)
+* Forums:
+  [crossplane-dev](https://groups.google.com/forum/#!forum/crossplane-dev)
+* Twitter: [@crossplane_io](https://twitter.com/crossplane_io)
+* Email: [info@crossplane.io](mailto:info@crossplane.io)
+
+## Governance and Owners
+
+provider-aws is run according to the same
```
Comment (luebken) [inline README.md:283]: terrajet

File: README.md [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+* [Crossplane: provider-terraform](https://github.com/crossplane-contrib/provider-terraform)
+* [Hashicorp Terraform Cloud Operator](https://github.com/hashicorp/terraform-k8s)
+* [Rancher Terraform Controller](https://github.com/rancher/terraform-controller)
+* [OAM Terraform Controller](https://github.com/oam-dev/terraform-controller)
+* [Kubeform](https://github.com/kubeform/kubeform)
+* [Terraform Operator](https://github.com/isaaguilar/terraform-operator)
+
+## Code of Conduct
+
+provider-aws adheres to the same [Code of
```
Comment (luebken) [inline README.md:304]: terrajet

File: README.md [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+Conduct](https://github.com/crossplane/crossplane/blob/master/CODE_OF_CONDUCT.md)
+as the core Crossplane project.
+
+## Licensing
+
+provider-aws is under the Apache 2.0 license.
```
Comment (luebken) [inline README.md:310]: terrajet

File: README.md [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+
+[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fcrossplane%2Fprovider-aws.svg?type=large)](https://app.fossa.io/projects/git%2Bgithub.com%2Fcrossplane%2Fprovider-aws?ref=badge_large)
+
+[design-doc]: https://github.com/crossplane/crossplane/blob/master/design/design-doc-terrajet.md
```
Comment (luebken) [inline README.md:314]: 404. Not merged Yet?

**Review by luebken — COMMENTED** [review]
Nice. I left some comments. In a future iteration, we should split up the readme so it only contains the overview.

**Comment by muvaf** [comment]
@luebken I've removed all the instructions since they are still changing fast, left only the description parts.

**Review by luebken — APPROVED** [review]
LGTM

**Comment by luebken** [comment]
Added https://github.com/crossplane-contrib/terrajet/issues/75 for the instructions.

**Review by turkenh — APPROVED** [review]

**Pull request merged**